File: Demo12/src/com/test/Test1.java
package com.test;

public class Test1 {

    public static void main(String[] args) {
       System.out.println( a(3));//5
    }

    public static int a(int b) {

        if (b <= 0) return 1;

        else {

            return a(b - 1) + a(b - 2);
        }

    }
}

File: Demo16/src/com/test/Test1.kt
package com.test

fun main(args: Array<String>) {
    val lambda = { left: Int, right: Int
        ->
        left + right
    }
    //后面我们用 () 的形式调用这个 Lambda 表达式，其实这个 () 对应的是 invoke 方法，换句话说，我们在这里也可以这么写：
    println(lambda(2, 3))
    println(lambda.invoke(2,3))
}